core/src/lib.rs

```diff
@@ -74,6 +74,42 @@ pub trait IsInBlockchain {
     fn is_in_blockchain(&self, wsv: &WorldStateView) -> bool;
 }
 
+/// Module constaining [`Starter`]
+pub mod handler {
+    #![allow(
+        clippy::expect_used,
+        clippy::std_instead_of_alloc,
+        clippy::panic,
+        clippy::print_stdout
+    )]
+    use std::thread::JoinHandle;
+
+    /// Call cancellation function and join thread on drop
+    pub struct ThreadHandler {
+        /// Shutdown function: after calling it thread must terminate in finite amount of time
+        shutdown: Option<Box<dyn FnOnce() + Send>>,
```

Comment:
What about to use [`fn pointers`](https://doc.rust-lang.org/reference/types/function-pointer.html)? They are safe and can be used here for a polymorphic behaviour.

Reply:
Can you motivate why they would be better in this situation? Why is a polymorphic behaviour desired?

Reply:
> What about to use [`fn pointers`](https://doc.rust-lang.org/reference/types/function-pointer.html)? They are safe and can be used here for a polymorphic behaviour.

As we discussed in DM, we want closure to capture objects (e.g. sender part of channel to send shutdown signal) unfortunately fn pointers incapable of this.

Reply:
> Can you motivate why they would be better in this situation? Why is a polymorphic behaviour desired?

After discussion in DM with @Erigara I'm agree, that using trait object is a good choice here.
However one pros of using fn pointers is to remove needless allocation. But with that approach we can't use objects, which implements `FnOnce` including move closures. And here in `Kura` we use such a closure.